maas-api.yaml

```diff
@@ -1,555 +1,677 @@
-                type: string
-                description: Region name/code from `regions.json`.
-              operatorID:
-                type: string
-                description: TSP ID from `info/operator.json`.
-              modes:
-                type: array
-                items:
-                  type: string
-                description: Public transit modes (pt_pub_*) for which results should be returned. [Mode identifier](https://developer.tripgo.com/faq/#mode-identifiers) strings like `pt_pub` or `pt_pub_subway`. In absence, all modes are included.
-              routesIDs:
-                type: array
-                items:
-                  type: string
-                  description: Route ID.
-                description: Route IDs to retrieve information from. In case of any missmatch, it'll return error response. Use either this filter or `routeNames`.
-              routesNames:
-                type: array
-                items:
-                  type: string
-                  description: Route name.
-                description: Route names to retrieve information from. In case of any missmatch, it'll return error response. Use either this filter or `routeIDs`.
-              onlyRealTime:
-                type: boolean
-                description: Boolean to filter only routes with Real Time support. When true, returned route list will contain only routes with Real Time service.
-                default: false
-              full:
-                type: boolean
-                description: Boolean if result should include full information for route/s or just a list with all their IDs.
-                default: false
-            required:
-              - region
-              - operatorID
-            example:
-              region: US_CA_LosAngeles
-              operatorID: LACMTA_Rail
-              modes:
-                - pt_pub_tram
-              full: true
+            type: string        
+      responses:
+        200:
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/RoutesResponse'
+
+  /agencies/{aid}/routes/{rid}/services:
+    get:
+      tags:
+        - Public Transport
+      summary: All services for a particular route
+      description:
+        Retrieves detailed information about services for a particular route
+      parameters:
+        - name: aid
+          in: path
+          required: true
+          description: Agency ID
+          schema:
+            type: string 
+        - name: rid
+          in: path
+          required: true
+          description: Route ID
+          schema:
+            type: string 
       responses:
         200:
           description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/RoutesResponse'
+
+  /agencies/{aid}/services/{sid}:
```

Comment:
can we move this to `/services` and have parameters for `aid` and `sid`?

Reply:
To keep it most consistent with other endpoints, this should rather be `GET /services/{sid}?agencyId={aid}

Reply:
Rational: Otherwise this would clash with `GET /services` when you provide agency ID + routes ID, which returns a different schema. Can't overload `GET` calls to a single endpoint like that, which would be a bit weird anyway.